lib/flatbuffers/flat_buffers.dart

```diff
@@ -126,17 +130,23 @@ class Builder {
 
   void _bufClear([int length]) {
     length ??= _buf.lengthInBytes;
-    // TODO these are slow, can we use memset() from DynamicLibrary.process()?
+    // Note: memset is faster than any of the following alternatives
+    _memset ??= DynamicLibrary.process().lookupFunction<
```

Comment:
1. makes sure to produce the same output as the upstream flatbuffers (see flatbuffers_test.dart which otherwise fails). Of course I can remove that test if you're sure data can be uninitialized... 
2. apparently it has some other impacts. Without the clear, of with just clearing the first 4 bytes as you suggested, some tests fail suddenly (queries on strings). Which made me have another look and I believe there's another bug in dart flatbuffers implementation, let's see... https://github.com/google/flatbuffers/pull/6390
3. Valgrind complains: `Conditional jump or move depends on uninitialised value(s)`... don't know where, though...
```
==355627== Conditional jump or move depends on uninitialised value(s)
==355627==    at 0x504A4E8: ??? (in /tmp/t)
==355627==    by 0x4FD245D: ??? (in /tmp/t)
==355627==    by 0x4FE19DF: ??? (in /tmp/t)
==355627==    by 0x4FE186B: ??? (in /tmp/t)
==355627==    by 0x5049DE4: ??? (in /tmp/t)
==355627==    by 0x506A6B5: ??? (in /tmp/t)
==355627==    by 0x506C4DC: ??? (in /tmp/t)
==355627==    by 0x5064464: ??? (in /tmp/t)
==355627==    by 0x5053268: ??? (in /tmp/t)
==355627==    by 0x5037B89: ??? (in /tmp/t)
==355627==    by 0x505339D: ??? (in /tmp/t)
==355627==    by 0x500C816: ??? (in /tmp/t)
```

Reply:
I don't get the question. I've answered to the best of my knowledge. I have no idea if/why flatbuffers internally need the buffer cleared. It may be the case they're relying on the values being zero (as with the linked PR), it may be not necessary at all :man_shrugging:

Reply:
this code **is** inside FlatBuffer Builder. C++ FBB's `clear()` is called `reset()` in dart's FBB. 

Yep, let's address in a call, seems like we aren't on the same page.

Reply:
conversation moved